pkg/monitor/cluster/cluster.go

```diff
@@ -119,6 +114,18 @@ func NewMonitor(log *logrus.Entry, restConfig *rest.Config, oc *api.OpenShiftClu
 	}, nil
 }
 
+func getHiveClientSet(hiveRestConfig *rest.Config) (client.Client, error) {
+	if hiveRestConfig == nil {
+		return nil, nil
+	}
+
+	hiveclientset, err := client.New(hiveRestConfig, client.Options{})
+	if err != nil {
+		return nil, err
```

Comment:
This 
```go
return nil, err
```
is the key to avoiding future panics as the comparison hiveclientset == nil will be evaluated to true (check the unit tests)